lib/nmatrix/math.rb

```diff
@@ -881,6 +880,41 @@ def nrm2 incx=1, n=nil
   end
   alias :norm2 :nrm2
 
+  #
+  # call-seq:
+  #     scale! -> NMatrix
+  #
+  # == Arguments
+  #   - +alpha+ -> Scalar value used in the operation.
+  #   - +inc+ -> Increment used in the scaling function. Should generally be 1.
+  #   - +n+ -> Number of elements of +vector+.
+  #
+  # This is a destructive method, modifying the source NMatrix.  See also #scale.
+  # Return the scaling result of the matrix. BLAS scal will be invoked if provided.
+
+  def scale!(alpha, incx=1, n=nil)
+    raise(DataTypeError, "Incompatible data type for the scaling factor") unless
+        NMatrix::upcast(self.dtype, NMatrix::min_dtype(alpha)) == self.dtype
+    return NMatrix::BLAS::scal(alpha, self, incx, self.size / incx) if NMatrix::BLAS.method_defined? :scal
```

Comment:
You are right. I will remove that section of ruby code.

Now the calling sequence is `scal <= BLAS::scal <= BLAS::cblas_scal <= Internal::BLAS::cblas_scal (if plugin not loaded) or ATLAS/LAPACKE::BLAS::cblas_scal (if loaded)`. Can it count as the exposure of BLAS `scal`? And as for the internal implementation of `cblas_scal`, is it exactly "a C variant of the BLAS method in NMatrix itself"?